library/src/test/java/com/datatorrent/lib/io/fs/HDFSOutputOperatorTest.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datatorrent.lib.io.fs;
+
+import java.io.IOException;
+
+import org.junit.Test;
+
+public class HDFSOutputOperatorTest extends AbstractFileOutputOperatorTest
+{
+
+  /**
+   * Test file rollover in case of idle windows
+   * 
+   * @throws IOException
+   */
+  @Test
+  public void testIdleWindowsFinalize() throws IOException
+  {
+    HDFSOutputOperator writer = new HDFSOutputOperator();
+    writer.setFileName("output.txt");
+    writer.setFilePath(testMeta.getDir());
+    writer.setAlwaysWriteToTmp(false);
```

Comment:
This would not trigger a file rolling since setAlwaysWriteToTmp is false.Could you please add a test case with it set to true and check the file rolling ? You can repeat both the test cases with it set to true and split the input across files and then check the results.
 + 1 for the test cases though, well thought of.